[CI] MatchFunctionIT class failing

**Build Scans:**
- [elasticsearch-intake #43947 / part3](https://gradle-enterprise.elastic.co/s/mmkbzpd6eb3lg)
- [elasticsearch-pull-request #164738 / part-3-fips-140-3](https://gradle-enterprise.elastic.co/s/j5mtmm3lnbqfq)
- [elasticsearch-pull-request #164738 / part-3](https://gradle-enterprise.elastic.co/s/lg45mm3app4ec)
- [elasticsearch-pull-request #164736 / part-3](https://gradle-enterprise.elastic.co/s/n5d5rhqkbrllg)
- [elasticsearch-pull-request #164735 / part-3](https://gradle-enterprise.elastic.co/s/feyu25pvsihuk)
- [elasticsearch-pull-request #164736 / part-3](https://gradle-enterprise.elastic.co/s/uwlgrzyxosivu)
- [elasticsearch-pull-request #164735 / releasetestcheckesqlengine_release-tests](https://gradle-enterprise.elastic.co/s/zxysjioaerdy4)
- [elasticsearch-pull-request #164735 / releasetestcheckesqlengine_release-tests](https://gradle-enterprise.elastic.co/s/yt2n2czlwtjyi)
- [elasticsearch-intake #43941 / part3](https://gradle-enterprise.elastic.co/s/gfty3vjb47nv2)
- [elasticsearch-pull-request #164717 / releasetestcheckesqlengine_release-tests](https://gradle-enterprise.elastic.co/s/vevuxh6cfxs7o)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.plugin.MatchFunctionIT.testWhereMatchWithScoringDifferentSort" -Dtests.seed=F9A925D166BA2591 -Dtests.locale=ckb -Dtests.timezone=Australia/Currie -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.plugin.MatchFunctionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.xpack.esql.plugin.MatchFunctionIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Request breaker not reset to 0 on node: node_s0
Expected: <0L>
     but: was <1048L>
```

**Issue Reasons:**
- [main] 3 consecutive failures in class org.elasticsearch.xpack.esql.plugin.MatchFunctionIT
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in step part-3-fips-140-3
- [main] 1 consecutive failures in step part-3
- [main] 2 consecutive failures in step releasetestcheckesqlengine_release-tests
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 1 consecutive failures in pipeline elasticsearch-pull-request
- [main] 10 failures in class org.elasticsearch.xpack.esql.plugin.MatchFunctionIT (4.0% fail rate in 250 executions)
- [main] 2 failures in step part3 (4.5% fail rate in 44 executions)
- [main] 4 failures in step part-3 (3.5% fail rate in 114 executions)
- [main] 3 failures in step releasetestcheckesqlengine_release-tests (6.4% fail rate in 47 executions)
- [main] 2 failures in pipeline elasticsearch-intake (4.5% fail rate in 44 executions)
- [main] 4 failures in pipeline elasticsearch-pull-request (3.2% fail rate in 126 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -610,12 +610,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.optimizer.PhysicalPlanOptimizerTests
   method: testMaxExpressionDepth_nestedAbs_minOverflow {default}
   issue: https://github.com/elastic/elasticsearch/issues/154341
-- class: org.elasticsearch.xpack.esql.plugin.MatchFunctionIT
-  method: testMatchInLookupJoinWithKeywordJoinField
-  issue: https://github.com/elastic/elasticsearch/issues/154357
-- class: org.elasticsearch.xpack.esql.plugin.MatchFunctionIT
-  method: testMatchRuntimeEvalWithIncompatibleIntegerValueThrowsError
-  issue: https://github.com/elastic/elasticsearch/issues/154359
 - class: org.elasticsearch.index.rankeval.RankEvalRequestIT
   method: testIndicesOptions
   issue: https://github.com/elastic/elasticsearch/issues/154360

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
@@ -861,6 +861,8 @@ public abstract class AbstractLookupService<R extends AbstractLookupService.Requ
                 searchContext.getSearchExecutionContext(),
                 QueryWarnings.NOOP
             );
+            // Queries built via the wrapper charge its own accounting pool, which nothing else drains.
+            searchContext.addReleasable(esqlCtx::releaseQueryConstructionMemory);
             return new LookupShardContext(
                 new EsPhysicalOperationProviders.DefaultShardContext(0, searchContext, esqlCtx, searchContext.request().getAliasFilter()),
                 esqlCtx,

